app/forms/application_form.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class ApplicationForm
+  include ActiveModel::Naming
+  include ActiveModel::Validations
+
+  # Overrides <tt>ActiveModel::Naming#model_name</tt> to take <tt>resource_name</tt>
+  # into account.
+  #
+  # @return [ActiveModel::Name]
+  def self.model_name
+    ActiveModel::Name.new(self, nil, resource_class.name)
+  end
+
+  # Sets or infers the resource name.
+  #
+  # @param resource [String] the name of the resource class
+  # @return [String]
+  # @example A class where the form object name does not match the resource:
+  #   class SignupForm < ApplicationForm
+  #     resource_name 'User'
+  #   end
+  def self.resource_class(klass = nil)
+    @resource_class ||= (klass || name.chomp('Form').constantize)
+  end
+
+  # The underlying model wrapped by the form.
+  #
+  # @return [Type<ApplicationRecord>]
+  attr_accessor :resource
+
+  delegate :id, :persisted?, to: :resource
+
+  def initialize(params = {})
```

Comment:
It seems like we need to support/enforce the passing of the actual resource into the initializer. Otherwise you can only use this base form object for new records and there would be no need to delegate those persistence-related methods. Is that right?